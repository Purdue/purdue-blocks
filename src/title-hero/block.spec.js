// title-hero test
jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
    selectOption,
} from '@wordpress/e2e-test-utils';

import {
    selectBlockByName,
    clickElementByText
} from '../test-helpers'

describe( 'Title Hero Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    // Tests can be added here by using the it() function
    it( 'Block should be available.', async () => {
        await insertBlock( 'Title Hero' )

        expect(await page.$('[data-type="purdue-blocks/title-hero"]')).not.toBeNull()

        expect( await getEditedPostContent()).toMatchSnapshot()
    } )

    it( 'Button should open Media Library.', async () => {
        await insertBlock( 'Title Hero' )
        await selectBlockByName( 'purdue-blocks/title-hero')

        // open media library
        await clickElementByText('button', 'Open Media Library')

        // media library title is there?
        expect(await page.$(`div.media-frame-title`)).not.toBeNull()
    })

    it( 'Title should be editable.', async () => {
        await insertBlock( 'Title Hero' )
        await selectBlockByName( 'purdue-blocks/title-hero')

        const typeString = "Test Title"
        
        await page.focus( `[data-type="purdue-blocks/title-hero"] input` );
        await page.keyboard.type(typeString, {delay: 10})

        const editedContent = await getEditedPostContent()

        expect( editedContent.includes(`<h1>${typeString}</h1>`)).toBe(true)
        expect( editedContent).toMatchSnapshot()
    })

    it( 'Intro copy should be editable.', async () => {
        await insertBlock( 'Title Hero' )
        await selectBlockByName( 'purdue-blocks/title-hero')

        const typeString = "Test intro copy here."
        
        await page.focus( `[data-type="purdue-blocks/title-hero"] textarea` );
        await page.keyboard.type(typeString, {delay: 10})

        const editedContent = await getEditedPostContent()

        expect( editedContent.includes(`<p>${typeString}</p>`)).toBe(true)
        expect( editedContent).toMatchSnapshot()
    })
} );
