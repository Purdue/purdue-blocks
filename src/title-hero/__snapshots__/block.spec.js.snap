// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Hero Block Block should be available. 1`] = `
"<!-- wp:purdue-blocks/title-hero -->
<div class=\\"pu-title-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1></h1><p></p></div></div></div></div></div>
<!-- /wp:purdue-blocks/title-hero -->"
`;

exports[`Title Hero Block Intro copy should be editable. 1`] = `
"<!-- wp:purdue-blocks/title-hero {\\"subText\\":\\"Test intro copy here.\\"} -->
<div class=\\"pu-title-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1></h1><p>Test intro copy here.</p></div></div></div></div></div>
<!-- /wp:purdue-blocks/title-hero -->"
`;

exports[`Title Hero Block Title should be editable. 1`] = `
"<!-- wp:purdue-blocks/title-hero {\\"pageTitle\\":\\"Test Title\\"} -->
<div class=\\"pu-title-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1>Test Title</h1><p></p></div></div></div></div></div>
<!-- /wp:purdue-blocks/title-hero -->"
`;
